vavr/src/main/java/io/vavr/control/Option.java

```diff
@@ -374,6 +427,39 @@ default T getOrElse(Supplier<? extends T> supplier) {
     }
 
     /**
+     * Maps the value and wraps it in a new {@code Some} if this is a {@code Some}, else returns {@code None}.
+     * In the case the given {@code mapper} throws, {@code None} will be returned.
+     * <p>
+     * Examples:
+     * <pre>{@code
+     * // = None
+     * Option.<Integer> none().mapTry(i -> 1 / i);
+     *
+     * // = Some(0)
+     * Option.some(2).mapTry(i -> 1 / i);
+     *
+     * // = None
+     * Option.some(0).mapTry(i -> 1 / i); // division by zero
+     * }</pre>
+     * 
+     * @param mapper A value mapper that may throw an exception
+     * @param <U>    The new value type
+     * @return a new {@code Some} containing the mapped value or {@code None}
+     */
+    default <U> Option<U> mapTry(CheckedFunction1<? super T, ? extends U> mapper) {
```

Comment:
Maybe `Option.tryToMap(...)` ? More natural and fluent.

Reply:
(see explanation above)